### DISCUSSION TITLE
After moving the layer in the editor, is there a callback provided?

### DISCUSSION THREAD

**Discussion by likaia** [hm21/pro_image_editor#263] [Q&A]
Sorry to bother you again. I moved the position of the layer in the canvas. Is there a corresponding callback function to get?

**Comment by hm21** [comment]
In that case, I recommend listening to `onUpdateLayer` as shown below.

```dart
  mainEditorCallbacks: MainEditorCallbacks(
      onUpdateLayer: (layer) {
        print(layer);
      },
      /// other callbacks
      onScaleStart: (value) {
        print(value);
      },
      onScaleUpdate: (value) {
        print(value);
      },
      onScaleEnd: (value) {
        print(value);
      },
    ),
```

**Reply by likaia** [reply]
OK，thinks.

I also want to ask if there is a callback after selecting the layer?

**Reply by hm21** [reply]
Currently, there is no callback triggered after selecting a layer. However, you can use the onTap callback to listen for every tap in the editor and then use a GlobalKey to check whether the layer is selected.
